### ISSUE TITLE
IPFS dependencies cause Yarn warnings

### ISSUE THREAD

**Issue by blixt** [jspm/generator#186]
This warning is output when installing `@jspm/generator`:

```
➤ YN0061: │ multiaddr@npm:10.0.1 is deprecated: This module is deprecated, please upgrade to @multiformats/multiaddr
➤ YN0061: │ multiaddr-to-uri@npm:8.0.0 is deprecated: This module is deprecated, please upgrade to @multiformats/multiaddr-to-uri
➤ YN0002: │ dns-over-http-resolver@npm:1.2.3 doesn't provide node-fetch (p89343), requested by native-fetch
➤ YN0002: │ ipfs-grpc-client@npm:0.9.4 doesn't provide google-protobuf (p11141), requested by @improbable-eng/grpc-web
```

It looks as if IPFS client 0.9.2 has moved on to at least avoid the first two warnings, though I'm not sure about the latter two. Might be worth trying to upgrade, or even making the IPFS functionality optional since I imagine it's not super commonly used?

**Comment by Aslemammad** [comment]
@guybedford I'm going to work on this! but shall we make the IPFS optional (with optional deps)?

**Comment by blixt** [comment]
Is there another release of this package scheduled soon, to make this fix available?

**Comment by guybedford** [comment]
Thanks for the ping, I've published the 1.0.0.
